src/main/java/com/hackclub/hccore/slack/SlackBot.java

```diff
@@ -209,6 +217,78 @@ public void onAfkChange(PlayerAFKStatusChangeEvent e) throws IOException {
         nowAfk ? playerAfkEnterAvatarUrl : playerAfkLeaveAvatarUrl, "AFK");
   }
 
+  @EventHandler
+  public void onAdvancementDone(PlayerAdvancementDoneEvent e) throws IOException {
+    PlayerData data = this.plugin.getDataManager().getData(e.getPlayer());
+    Advancement advancement = e.getAdvancement();
+
+    // Filter out hidden advancements and recipes
+    if (advancement.getDisplay() == null) {
+      return;
+    }
+    if (!advancement.getDisplay().doesAnnounceToChat()) {
+      return;
+    }
+
+    String advancementName = PlainTextComponentSerializer.plainText()
+        .serialize(advancement.getDisplay().title());
+
+    String advancementType;
+    switch (advancement.getDisplay().frame()) {
+      case GOAL ->
+          advancementType = "goal";
+      case CHALLENGE ->
+          advancementType = "challenge";
+      default ->
+          advancementType = "advancement";
+    }
+
+    sendMessage(
+        "%s has completed the %s *%s*".formatted(data.getUsableName(), advancementType, advancementName),
+        playerAdvancementAvatarUrl, "Advancement");
+  }
+
```

Comment:
can this repeated code be refactored? maybe not needed as its only twice.
does not block merging, only if you feel its needed

Reply:
i think it doesn't really need to be refactored, i don't see a big benefit to doing so.